src/optimise.jl

```diff
@@ -1,17 +1,37 @@
-
+﻿
 """
-    minimise(f, X, tol=1e-3)
+    minimise(f, X, structure = SortedArray, tol=1e-3) 
+    or
+    minimise(f, X, structure = HeapedArray, tol=1e-3)
+    or 
+    minimise(f, X, tol=1e-3) in this case the default value of "structure" is "HeapedArray"
 
-Find the global minimum of the function `f` over the `Interval` or `IntervalBox` `X` using the Moore-Skelboe algorithm.
+Find the global minimum of the function `f` over the `Interval` or `IntervalBox` `X` using the Moore-Skelboe algorithm. By specifing the way in which vector element are kept arranged which is in heaped array or in sorted array. If you not specify any particular strategy to keep vector elements arranged then by default heaped array is used.  
 
 For higher-dimensional functions ``f:\\mathbb{R}^n \\to \\mathbb{R}``, `f` must take a single vector argument.
 
 Returns an interval containing the global minimum, and a list of boxes that contain the minimisers.
 """
-function minimise(f, X::T, tol=1e-3) where {T}
 
-    # list of boxes with corresponding lower bound, ordered by increasing lower bound:
-    working = SortedVector([(X, ∞)], x->x[2])
+export HeapedArray,SortedArray
+
+abstract type Strategy end
+struct H <:Strategy end
+struct S <:Strategy end
+
+const HeapedArray = H()
+const SortedArray = S()
+
+DefStructure=HeapedArray
+
+function minimise(f, X::T ;structure::Strategy=DefStructure ,tol=1e-3 ) where {T}
+
+    # list of boxes with corresponding lower bound, arranged according to selected structure :
+    if structure==SortedArray
+       working = SortedVector([(X, ∞)], x->x[2])
```

Comment:
Please change this in the way I requested.

Reply:
ok..!!